compiler/rustc_next_trait_solver/src/solve/trait_goals.rs

```diff
@@ -111,6 +111,15 @@ where
             if trait_clause.def_id() == goal.predicate.def_id()
                 && trait_clause.polarity() == goal.predicate.polarity
             {
+                if !DeepRejectCtxt::new(ecx.cx(), TreatParams::AsRigid, TreatParams::AsRigid)
+                    .args_may_unify(
+                        goal.predicate.trait_ref.args,
+                        trait_clause.skip_binder().trait_ref.args,
+                    )
+                {
+                    return Err(NoSolution);
+                }
+
```

Comment:
please also add this for `NormalizesTo` goals

Reply:
done